src/Language/PureScript/Docs/Prim.hs

```diff
@@ -377,70 +377,55 @@ coercible :: Declaration
 coercible = primClassOf (P.primSubName "Coerce") "Coercible" $ T.unlines
   [ "Coercible is a two-parameter type class that has instances for types `a`"
   , "and `b` if the compiler can infer that they have the same representation."
-  , "This class does not have regular instances; instead they are created"
-  , "on-the-fly during type-checking according to a set of rules."
+  , "Coercible constraints are solved according to the following rules:"
   , ""
-  , "First, Coercible obeys reflexivity - any type has the same representation"
-  , "as itself:"
+  , "* _reflexivity_, any type has the same representation as itself:"
+  , "`Coercible a a` holds."
   , ""
-  , "    instance coercibleReflexive :: Coercible a a"
+  , "* _symmetry_, if a type `a` can be coerced to some other type `b`, then `b`"
+  , "can also be coerced back to `a`: `Coercible a b` implies `Coercible b a`."
   , ""
-  , "Second, Coercible obeys symmetry - if a type `a` can be coerced to some"
-  , "other type `b`, then `b` can also be coerced back to `a`:"
+  , "* _transitivity_, if a type `a` can be coerced to some other type `b` which"
+  , "can be coerced to some other type `c`, then `a` can also be coerced to `c`:"
+  , "`Coercible a b` and `Coercible b c` imply `Coercible a c`."
   , ""
-  , "    instance coercibleSymmetric :: Coercible a b => Coercible b a"
+  , "* Newtypes can be freely wrapped and unwrapped when their constructor is"
+  , "in scope:"
   , ""
-  , "Third, Coercible obeys transitivity - if a type `a` can be coerced to some"
-  , "other type `b` which can be coerced to some other type `c`, then `a` can"
-  , "also be coerced to `c`:"
+  , "      newtype Age = Age Int"
   , ""
-  , "    instance coercibleTransitive :: (Coercible a b, Coercible b c) => Coercible a c"
+  , "`Coercible Int Age` and `Coercible Age Int` hold since `Age` has the same"
+  , "runtime representation than `Int`."
   , ""
-  , "Fourth, for every type constructor there is an instance that allows one"
-  , "to coerce under the type constructor (`data` or `newtype`). For example,"
-  , "given a definition:"
+  , "Newtype constructors have to be in scope to preserve abstraction. It's"
+  , "common to declare a newtype to encode some invariants (non emptiness of"
+  , "arrays with `Data.Array.NonEmpty.NonEmptyArray` for example), hide its"
+  , "constructor and export smart constructors instead. Without this restriction,"
+  , "the guarantees provided by such newtypes would be void."
   , ""
-  , "data D a b = D a"
+  , "* Two types of kind `Type` may be coercible, but only if their heads are"
```

Comment:
I'm a bit uneasy with this explanation for another reason beyond the way "head" is used here (which I would regard as incorrect, see my other comment), but also because I think it only works if you forget about newtypes. For example, `Int` and `Age` are both types of kind `Type` which are not the same,  but they are still coercible.

Reply:
Maybe (with proper line reformatting of course) this?

```suggestion
  , "* If none of the above are applicable, two types of kind `Type` may be coercible, but only if their heads are"
```

Reply:
Ah good idea, I think that works!

Reply:
I applied your suggestion.